sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageBatch.cs

```diff
@@ -118,15 +118,15 @@ public override bool TryAddMessage(ServiceBusMessage message)
                         ? OverheadBytesSmallMessage
                         : OverheadBytesLargeMessage);
 
-                if (size > MaxSizeInBytes)
+                if (size > MaxSizeInBytes || BatchMessages.Count >= Options.MaxSizeInNumberOfItems.Value)
                 {
                     return false;
                 }
 
                 _sizeBytes = size;
                 BatchMessages.Add(message);
 
-            return true;
+                return true;
```

Comment:
Can be reverted if not wanted. It just looked awkward and I felt it is not a big sin to change this. Also happy to pull it into a dedicated commit if prefered